resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/resources/undo/snapshot/AbstractResourceSnapshot.java

```diff
@@ -0,0 +1,140 @@
+/*******************************************************************************
+ * Copyright (c) 2006, 2023 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.core.resources.undo.snapshot;
+
+import org.eclipse.core.resources.IContainer;
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.resources.ResourceAttributes;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+
+/**
+ * Base implementation of ResourceSnapshot that describes the common attributes
+ * of a resource to be created.
+ *
+ * This class is not intended to be instantiated or used by clients.
```

Comment:
We should maybe remove this line?

Reply:
This line should not be removed. The approved way of accessing these classes is by calling ResourceSnapshot.fromResource(r);

Reply:
So couldn't the constructors be only package-visible? And the documentation mention that the right approach to get an instance if `ResourceSnapshot.fromResource()` ?